### DISCUSSION TITLE
Anyone having problem with overwriting the :root variables?

### DISCUSSION THREAD

**Discussion by jkolski** [twbs/bootstrap#37787] [Q&A] [answered]
Hi all,

I'm trying to use CDN hosted Bootstrap but force eg. my colors in my own CSS file. It looks kinda like this:
```
...
<html ...>
  <head>
...
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css" rel="stylesheet" ...>
    <link href="/assets/css/main.css" rel="stylesheet">
...
```
and the `main.css` file contains my variables with identical names like those in BS:
```
:root {
  --bs-dropdown-link-active-bg: #FF0000; /* not the default blue */
}
```
but somehow my browser (Chrome, macOS) forces using the variable from the `_dropdown.scss`:
<img width="379" alt="image" src="https://user-images.githubusercontent.com/988380/210284606-57194f10-7b4e-4cea-9040-60803ef6fc2e.png">
and my `:root` settings are crossed out, even with the `!importang` flag:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/988380/210284513-8d1d5812-de65-4501-9bff-978d96dee2bd.png">

I know I can download the entire project and change colors in the sass and recompile entire project or I can find all classes using some color and then add them as selectors to my CSS file, but I was thinking about much less entertaining way of spending my evenings ;) I found that it works the same when I download the full CSS file and remove link to `bootstrap.min.css.map`. Also doesn't work even if I'll declare variables in the `<style>` tag directly in head.

Anybody has an idea why Chrome is prioritizing the file from map file over my code? My declaration is later then the Bootstrap file, it has `!important` flag and stuff :(

Best,
Jakub

**Comment by mdo** [comment, marked as answer]
Variables for components are set on the component level. Instead of this:

```html
:root {
  --bs-dropdown-link-active-bg: #FF0000; /* not the default blue */
}
```

You need this:

```html
.dropdown-menu {
  --bs-dropdown-link-active-bg: #FF0000;
}
```

This is mentioned in every component's documentation, like here: https://getbootstrap.com/docs/5.3/components/dropdowns/#variables.
